exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -190,17 +193,25 @@ export function createCacheKey (pkgs: string[], registries: Record<string, strin
   return createBase32Hash(hashStr)
 }
 
-function isCacheValid (cacheLink: string, dlxCacheMaxAge: number): boolean {
+function getValidCacheDir (cacheLink: string, dlxCacheMaxAge: number): string | undefined {
   let stats: Stats
+  let target: string
   try {
     stats = fs.lstatSync(cacheLink)
+    if (stats.isSymbolicLink()) {
+      target = fs.realpathSync(cacheLink)
+      if (!target) return undefined
```

Comment:
Is this line necessary?

Reply:
no, but it is weird to allow the validity check below if the target doesn't exist.